calendarpal/src/main/java/org/dmfs/android/calendarpal/predicates/CalendarScoped.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.android.calendarpal.predicates;
+
+import android.provider.CalendarContract;
+import android.support.annotation.NonNull;
+
+import org.dmfs.android.contentpal.Predicate;
+import org.dmfs.android.contentpal.RowSnapshot;
+import org.dmfs.android.contentpal.predicates.AllOf;
+import org.dmfs.android.contentpal.predicates.DelegatingPredicate;
+import org.dmfs.android.contentpal.predicates.ReferringTo;
+
+
+/**
+ * {@link Predicate} decorator to add selection for rows referring to the given calendar.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class CalendarScoped extends DelegatingPredicate
+{
+    public CalendarScoped(RowSnapshot<CalendarContract.Calendars> calendarRow, @NonNull Predicate predicate)
+    {
+        super(new AllOf(predicate, new ReferringTo<>(CalendarContract.Events.CALENDAR_ID, calendarRow)));
```

Comment:
Please add a quick unit test.

Reply:
I've added a unit test. Note that I couldn't use the `Values` and other helper classes from `contentpal` test classpath, not sure where to put them to be able to use them in other modules as well.

Reply:
The predicate tesing utils in `contentpal` should go into `contentpal_testing` as well to make them available to all other modules. Please create an issue and PR.

Reply:
That was my first move but `Mocked` uses `ValueArgument` which is not from `contentpal-api` so I thought it won't work but now I see that that's actually the only class from the implementation and it's a trivial one, so I guess it's okay to re-implement that in the test module.
I am going to create the issue and PR.